mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -540,6 +536,21 @@ class Word2VecModel private[mllib] (
       (word, wordVectors.slice(vectorSize * ind, vectorSize * ind + vectorSize))
     }
   }
+
+  /**
+   * Euclidean Normalization for a vector
+   * @param vector An array to be normalized
+   * @return a new normalized array
+   */
+  def euclideanNormalize(vector: Array[Float]):Array[Float] = {
+    val norm = blas.snrm2(vector.size, vector, 1)
+
+    if (norm == 0) {
+      Array.fill[Float](vector.size)(0)
+    } else {
+      vector.map(v => v/norm)
```

Comment:
Good point. I have incorporated this in the new PR: https://github.com/apache/spark/pull/6245